Tests/Unit/Domain/Schema/DtoSpecificationTest.php

```diff
@@ -0,0 +1,175 @@
+            'isSupported' => false,
+            'isDTO' => false,
+            'isDTC' => false,
+            'isSVDTO' => false,
+        ];
+
+        yield Fixtures\InvalidObjects\CollectionThatIsNotReadonly::class => [
+            'typeName' => Fixtures\InvalidObjects\CollectionThatIsNotReadonly::class,
+            'isSupported' => false,
+            'isDTO' => false,
+            'isDTC' => false,
+            'isSVDTO' => false,
+        ];
+
+        // invalid objects
+
+        yield Fixtures\InvalidObjects\ObjectThatIsNotReadonly::class => [
+            'typeName' => Fixtures\InvalidObjects\ObjectThatIsNotReadonly::class,
+            'isSupported' => false,
+            'isDTO' => false,
+            'isDTC' => false,
+            'isSVDTO' => false,
+        ];
+
+        yield Fixtures\InvalidObjects\ObjectThatHasNotSupportedProperties::class => [
+            'typeName' => Fixtures\InvalidObjects\ObjectThatHasNotSupportedProperties::class,
+            'isSupported' => false,
+            'isDTO' => false,
+            'isDTC' => false,
+            'isSVDTO' => false,
+        ];
+
+        yield Fixtures\InvalidObjects\ObjectThatHasNonPromotedConstructorArguments::class => [
+            'typeName' => Fixtures\InvalidObjects\ObjectThatHasNonPromotedConstructorArguments::class,
+            'isSupported' => false,
+            'isDTO' => false,
+            'isDTC' => false,
+            'isSVDTO' => false,
+        ];
+
+        yield Fixtures\InvalidObjects\NonBackedEnum::class => [
+            'typeName' => Fixtures\InvalidObjects\NonBackedEnum::class,
+            'isSupported' => false,
+            'isDTO' => false,
+            'isDTC' => false,
+            'isSVDTO' => false,
+        ];
+    }
+
+    /**
+     * @dataProvider isSupportedInDataProvider
+     * @param class-string $className
+     */
+    public function testIsSupportedIn(string $className, bool $isSupported, bool $isDTO, bool $isDTC, bool $isSVDTO): void
+    {
+        $this->assertEquals($isSupported, IsSupportedInSchema::isSatisfiedByClassName($className), sprintf('IsSupportedInSchema should return "%s" for class "%s"', $isSupported ? 'true' : 'false', $className));
+        $this->assertEquals($isDTO, IsDataTransferObject::isSatisfiedByClassName($className), sprintf('IsDataTransferObject should return "%s" for class "%s"', $isSupported ? 'true' : 'false', $className));
+        $this->assertEquals($isDTC, IsDataTransferObjectCollection::isSatisfiedByClassName($className), sprintf('IsDataTransferObjectCollection should return "%s" for class "%s"', $isSupported ? 'true' : 'false', $className));
+        $this->assertEquals($isSVDTO, IsSingleValueDataTransferObject::isSatisfiedByClassName($className), sprintf('IsSingleValueDataTransferObject should return "%s" for class "%s"', $isSupported ? 'true' : 'false', $className));
+    }
+
+    public function testFoo(): void
```

Comment:
This is obsolete because already covered in the last test case from above, right?

Reply:
Correct. During cleanup this was too far at the end to be noticed.

Reply:
removed the obsolete test will merge now